src/app/api/following/route.ts

```diff
@@ -45,3 +45,139 @@ export async function GET(req: NextRequest) {
     });
   }
 }
+export async function POST(req: NextRequest) {
+  try {
+    const db = connectDB();
+    const url = new URL(req.url ? req.url : "invalid");
+    const follower = url.searchParams.get("username");
+    // console.log("backend post:", follower);
+
+    if (!db) {
+      throw new Error("Failed to connect to the database");
+    }
+
+    const { following } = await req.json();
+    // console.log("follower: ", follower);
+    // console.log("following: ", following);
+    // make sure FOLLOWER exist
+    const { data: check_follower, error: follower_exists_error } = await db
+      .from("users")
+      .select("*")
+      .match({ username: follower });
+    // console.log("check follower: ", check_follower);
+
+    if (!check_follower || check_follower.length === 0) {
+      throw new Error("follower input invalid: user not found");
+    }
+    if (follower_exists_error) {
+      throw follower_exists_error;
+    }
+    //make sure FOLLOWING exists
+    const { data: check_following, error: following_exists_error } = await db
+      .from("users")
+      .select("*")
+      .match({ username: following });
+    // console.log("check_following", check_following);
+
+    if (!check_following || check_following.length === 0) {
+      throw new Error("data input invalid: user not found");
+    }
+    if (following_exists_error) {
+      throw following_exists_error;
+    }
+
+    // make sure relationship isn't already in the database (can't follow more than once)
+    const { data: check_rel, error: rel_exists_error } = await db
+      .from("following")
+      .select("*")
+      .match({ follower: follower, following: following });
+    // console.log("check_relationship", check_rel);
+
+    if (!check_rel || check_rel.length > 0) {
+      throw new Error("relationship already exists");
+    }
+    if (rel_exists_error) {
+      throw rel_exists_error;
+    }
+    //no issues, insert the friendship
+    const { data: following_info, error } = await db
+      .from("following")
+      .insert([{ follower, following }] as any);
+
+    if (error) {
+      throw error;
+    }
+
+    return NextResponse.json({
+      message: "Success",
+      status: 201,
+    });
+  } catch (error) {
+    console.error(error); // Log the error to the console
+    return NextResponse.json({
+      message: `No friend for you`,
+      status: 500,
+    });
+  }
+}
+
+export async function DELETE(req: Request) {
+  try {
+    const db = connectDB();
+    const url = new URL(req.url ? req.url : "invalid");
+    const follower = url.searchParams.get("username");
+    // console.log("backend post:", follower);
+
+    if (!db) {
+      throw new Error("Failed to connect to the database");
+    }
+
+    const { following } = await req.json();
+    // console.log("follower: ", follower);
```

Comment:
We'll have to clean up the comments/logs later on

Reply:
Thank you for pointing that out I will definitely go through and clean up these parts once we finish. For now I'm going to keep them as I continue to use those statements to see what is happening at each point in the code.